src/components/Pagination/Pagination.st.css

```diff
@@ -0,0 +1,264 @@
+.pageButtonRules {
+  min-width: 1em;
+  min-height: 1em;
+  line-height: 1em;
+  justify-content: center;
+  align-items: center;
+  padding: 0.5em 0.5em;
+  margin: 0.25em;
+}
+
+.root {
+  -st-extends: Pagination;
+  -st-states: rtl, paginationMode(enum(pages, input)), navButtonsType(enum(text, arrows)), navButtonsAmount(number), alignment(enum(center, left, right));
+  align-items: stretch;
+  box-sizing: border-box;
+  display: flex;
+  font:  wixApply(fallback, wixApply(font, value(TextFont)), value(fnt));
+}
+
+.root:alignment(left) {
+  justify-content: flex-start;
+}
+
+.root:alignment(center) {
+  justify-content: center;
+}
+.root:alignment(right) {
+  justify-content: flex-end;
+}
+
+.root::pageButton {
+  -st-mixin: pageButtonRules;
+  position: relative;
+  color: wixApply(color, wixApply(fallback, value(TextFontColor), value(pageNumbersTextColor)));
+}
+
+.root::currentPage {
+  -st-mixin: pageButtonRules;
+  color: wixApply(color, wixApply(fallback, value(SelectedTextFontColor), value(selectedPageTextColor)));
+  background: value(selectedPageBG);
+  border-radius: value(bRadius);
+  border: value(selectedPageBorderWidth) solid value(selectedPageBorderColor);
+  box-shadow: value(shadow);
+  padding: calc(0.5em - value(selectedPageBorderWidth)) calc(0.5em - value(selectedPageBorderWidth));
+}
+
+.root:not(:disabled)::pageButton:hover::before {
+  background: value(selectedPageBG);
+  opacity: 0.2;
+  content: "";
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  border-radius: value(bRadius);
+}
+
+.root:not(:disabled)::pageButton:hover {
+  opacity: 0.7;
+}
+
+.root:disabled::pageButton {
+  cursor: default;
+}
+
+.root::navButton {
+  min-width: 1em;
+  min-height: 1em;
+  padding: calc(0.75em - value(navButtonBorderWidth));
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border: value(navButtonBorderWidth) solid value(navButtonBorderColor);
+  background: value(navButtonBG);
+  border-radius: value(bRadius);
+}
+
+.root::navButton:disabled, .root[data-preview="disabled"]::navButton {
+  padding: calc(0.75em - value(navButtonDisabledBorderWidth));
+  border: value(navButtonDisabledBorderWidth) solid value(navButtonDisabledBorderColor);
+  background: value(navButtonDisabledBG);
+  cursor: default;
+}
+
+.root::navButton:disabled svg, .root[data-preview="disabled"]::navButton svg {
+  fill: value(arrowsDisabledColor);
+}
+
+.root::navButton:hover:not(:disabled) {
+  opacity: 0.8;
+}
+
+.root::navButton svg {
+  width: 1em;
+  min-width: 16px;
+  height: 1em;
+  min-height: 16px;
+  fill: value(arrowsColor);
+}
+
+.root[dir="rtl"]::navButton svg {
+  transform: scaleX(-1);
+}
+
+.root::navButton span {
+  font-size: value(navButtonTextSize);
+  color: value(navButtonTextColor);
+  line-height: 1em;
+}
+
+.root::navButton:disabled span, .root[data-preview="disabled"]::navButton span {
+  color: value(navButtonDisabledTextColor);
+}
+
+.root::navButtonFirst {
+  margin-right: 1em;
```

Comment:
Again, `rtl` might be a problem here:
Either we use `margin-right` but the first `navButton` will always appear on the left side (meaning not using `flex`), or using `flex`, and having an empty `div` with `width: 1em` or something like that.